src/modules/visit-requests/hooks/get-active-visit-for-user-and-space.ts

```diff
@@ -1,23 +1,44 @@
-import { useQuery, UseQueryResult } from '@tanstack/react-query';
+import { QueryClient, useQuery, UseQueryResult } from '@tanstack/react-query';
 
 import { User } from '@auth/types';
-import { VisitsService } from '@modules/visit-requests/services';
 import { QUERY_KEYS } from '@shared/const/query-keys';
 import { Visit } from '@shared/types';
+import { VisitsService } from '@visit-requests/services';
+
+export async function getActiveVisitForUserAndSpace(
+	visitorSpaceSlug: string,
+	user: User | null | undefined
+): Promise<Visit | null> {
+	if (!user) {
+		console.log('visit request getActiveVisitForUserAndSpace', null);
+		return null; // Anonymous users can never have an active visit request
+	}
+	return VisitsService.getActiveVisitForUserAndSpace(visitorSpaceSlug);
+}
 
 export function useGetActiveVisitForUserAndSpace(
 	visitorSpaceSlug: string,
 	user: User | null | undefined,
-	enabled = true
+	options: { enabled?: boolean } = {}
 ): UseQueryResult<Visit> {
+	console.log('getting active visit request', { visitorSpaceSlug, user });
 	return useQuery(
-		[QUERY_KEYS.getActiveVisitForUserAndSpace, { spaceId: visitorSpaceSlug, user }],
-		() => {
-			if (!user) {
-				return null; // Anonymous users can never have an active visit request
-			}
-			return VisitsService.getActiveVisitForUserAndSpace(visitorSpaceSlug);
-		},
-		{ enabled, retry: 0 }
+		[QUERY_KEYS.getActiveVisitForUserAndSpace, visitorSpaceSlug, user?.id || null],
+		() => getActiveVisitForUserAndSpace(visitorSpaceSlug, user || null),
+		{ enabled: true, retry: 0, ...options }
+	);
+}
+
+export async function makeServerSideRequestGetActiveVisitForUserAndSpace(
+	queryClient: QueryClient,
+	visitorSpaceSlug: string
+): Promise<void> {
+	queryClient.setQueryData(
+		[QUERY_KEYS.getActiveVisitForUserAndSpace, visitorSpaceSlug, null],
+		null
 	);
+	// return queryClient.prefetchQuery({
+	// 	queryKey: [QUERY_KEYS.getActiveVisitForUserAndSpace, visitorSpaceSlug, null],
+	// 	queryFn: () => getActiveVisitForUserAndSpace(visitorSpaceSlug, null),
+	// });
```

Comment:
Uncomment?

Reply:
its replaced by the setQueryData since its more efficient
i'll remove the comments